local_planner/src/nodes/local_planner_node.cpp

```diff
@@ -87,17 +87,18 @@ void LocalPlannerNode::startNode() {
 
 void LocalPlannerNode::readParams() {
   // Parameter from launch file
-  auto goal = toPoint(local_planner_->getGoal());
-  nh_.param<double>("goal_x_param", goal.x, 9.0);
-  nh_.param<double>("goal_y_param", goal.y, 13.0);
-  nh_.param<double>("goal_z_param", goal.z, 3.5);
+  float init_goal_position_x, init_goal_position_y, init_goal_position_z;
+
+  nh_.param<float>("goal_x_param", init_goal_position_x, 9.0);
```

Comment:
This would be neater as just:
```suggestion
  nh_.param<float>("goal_x_param", goal_position_.x(), 9.0);
```
For x,y,z
Then you don't need the temporary variables.